### ISSUE TITLE
Do I have chance to get the real depth with the known focal length?

### ISSUE THREAD

**Issue by placeforyiming** [colmap/colmap#701]
If I choose the simple_pinhole model, give it the 

--ImageReader.default_focal_length_factor=focal_len/longest_edge

Then, run: 

colmap mapper  --database_path =./Generated_results/
                           --image_path=image_dir
                           --output_path=./Generated_results/
                           --Mapper.ba_refine_focal_length=0 
                          --Mapper.ba_refine_extra_params=0 

Should I get the real depth instead the relative depth?

I think so, but the depth map I got still needs a scale factor to match the ground truth.

**Comment by placeforyiming** [comment]
I apply colmap on different videos captured with the same camera, but the estimated depth for each video have different scale factors.

**Comment by abenbihi** [comment]
Since you seem to be using a single monocular camera, you can not get the real depth if you don't manually provide a scale.

**Comment by tsattler** [comment]
No you won't. Specifying a known focal length defines how points in normalized device coordinates (obtained after the perspective projection of 3D points into the normalized image plane at z=1) map to points in the image. Changing the scale of the scene does not change the projection of the 3D points into the normalized image plane.

**Comment by placeforyiming** [comment]
Thanks for all your reply!
@tsattler @abenbihi 
I made it work by calculating the scale factor from my own estimated pose and pose provided by colmap.

**Comment by placeforyiming** [comment]
Reopen it since I found I need a constant factor around 1.2 to map the depth map to real depth.

Where does this 1.2 come from?

I know the default focal_len/max(height,width)=1.2, but I have already changed this value according to real focal length.

I got depth map from colmap, re-scale it by using real_translation/colmap_translation for each sub-folder in dense. Then, I found I still need multiply 1.2, and this 1.2 works for different videos with different camera models.  It seems like 

Very curious where does this 1.2 come from?
